packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,595 @@
+	 * state in sync, the component has a 'state storage' that is updated when the
+	 * current state is changed due to a user action.
+	 *
+	 * Check if the previously saved state is outdated. That would mean
+	 * a property has been changed programatically because the previous state
+	 * is always updated in the interaction handlers.
+	 *
+	 * Will return true if any of the properties is not equal to its previously
+	 * stored value.
+	 *
+	 * @protected
+	 */
+	isCurrentStateUpdated() {
+		return Object.entries(this._stateStorage).some(([propName, propValue]) => this[propName] !== propValue);
+	}
+
+	/**
+	 * Returns the last stored value of a property
+	 *
+	 * @protected
+	 */
+	getStoredPropertyState(property) {
+		return this._stateStorage[property];
+	}
+
+	/**
+	 * Check if one or more properties have been updated compared to their last
+	 * saved values in the state storage.
+	 *
+	 * @protected
+	 */
+	isPropertyUpdated(...properties) {
+		return properties.some(prop => this.getStoredPropertyState(prop) !== this[prop]);
+	}
+
+	/**
+	 * Updates the previously saved in the _stateStorage values of one or more properties.
+	 *
+	 * @protected
+	 */
+	storePropertyState(...props) {
+		props.forEach(property => {
+			this._stateStorage[property] = this[property];
+		});
+	}
+
+	/**
+	 * Calculates and draws the tickmarks with a CSS gradient style
+	 *
+	 * @private
+	 */
+	_drawDefaultTickmarks(step, max, min) {
+		if (!this.tickmarks || !this.step) {
+			return;
+		}
+
+		// Convert number values to strings to let the CSS do calculations better
+		// rounding/subpixel behavior" and the most precise tickmarks distribution
+		const maxStr = String(max);
+		const minStr = String(min);
+		const stepStr = String(step);
+		const tickmarkWidth = "1px";
+
+		// There is a CSS bug with the 'currentcolor' value of a CSS gradient that does not
+		// respect the variable for more than one theme. It has to be set here for now.
+		const currentTheme = getTheme();
+		const currentColor = this.TICKMARK_COLOR_MAP[currentTheme];
+
+		this._tickmarksAmount = `${maxStr - minStr} / ${stepStr}`;
+		this._hiddenTickmarks = false;
+
+		// Transparent CSS gradient background
+		const tickmarksGradientBase = `linear-gradient(to right, ${currentColor} ${tickmarkWidth}, transparent 0) `;
+
+		// Draw the tickmarks as a patern over the gradient background
+		const tickmarksGradientdPattern = `0 center / calc((100% - ${tickmarkWidth}) / (${this._tickmarksAmount})) 100% repeat-x`;
+
+		// Combine to get the complete CSS background gradient property value
+		this._tickmarksBackground = `${tickmarksGradientBase + tickmarksGradientdPattern}`;
+
+		// If labelsInterval is specified draw labels for the necessary tickmarks
+		if (this.labelInterval > 0) {
+			this._drawDefaultLabels(parseInt(tickmarkWidth));
+		}
+	}
+
+	/**
+	 * Calculates the labels amout, width and text and creates them
+	 *
+	 * @private
+	 */
+	_drawDefaultLabels(tickmarkWidth) {
```

Comment:
Is tickmarkWidth used in the function, I could not see it, but I might missed it?